### ISSUE TITLE
send_meta() called with NULL pointer!

### ISSUE THREAD

**Issue by pille** [gsliepen/tinc#152]
this morning we had router-upgrades that created (partial) network-loss of our tinc-daemons accross data-centers. some of them (8/60) failed and died with ```send_meta() called with NULL pointer!```.

we're using old distribution versions:
```
tinc version 1.0.16
tinc version 1.0.26
```

i've checked the changelog, but could not find relevant commits in never releases.
the most recent commit referring to "NULL pointer" is e913f3f232e4809b7218d081ab9f94cef1c94ac3 (1.0.24)

**Comment by gsliepen** [comment]
What are the log messages leading up to the NULL pointer message? That would help in finding the cause of this.

**Comment by pille** [comment]
i don't think they are related or lead to anything:
```
Sep 28 03:54:02 tmpa tinc.VPN[15234]: Bogus data received from <unknown> (127.0.0.1 port 50094)
Sep 28 03:54:26 tmpa tinc.VPN[15234]: Flushing meta data to tmpb ($IP port 655) failed: Connection reset by peer
Sep 28 03:55:01 tmpa tinc.VPN[15234]: Bogus data received from <unknown> (127.0.0.1 port 50130)
Sep 28 03:56:01 tmpa tinc.VPN[15234]: Bogus data received from <unknown> (127.0.0.1 port 50156)
Sep 28 03:57:01 tmpa tinc.VPN[15234]: Bogus data received from <unknown> (127.0.0.1 port 50174)
Sep 28 03:57:44 tmpa tinc.VPN[15234]: send_meta() called with NULL pointer!
```

between each tinc logline, there are several from iptables dropping packages (unrelated ports), indicating network was still up at that point in time.
the exception was triggered when the nodes lost its uplink.

**Comment by estebannagy** [comment]
Hi, have the same problem. Lots of  bogus data, and then appears the NULL pointer! error, have to restart the VPN daemon. @pille did you resolve this?

**Comment by pille** [comment]
no, this happens everytime the hoster has a bigger networking problem.
so after that is fixed, the VPN still is stuck.
though i noticed that after the VPN is in a broken state, there's still traffic (TCP/655) exchanged between the nodes, but many packets hit the "invalid state" iptables filter.
